channeld/channeld.c

```diff
@@ -569,8 +650,54 @@ static void handle_peer_announcement_signatures(struct peer *peer, const u8 *msg
 			    type_to_string(tmpctx, struct channel_id, &chanid));
 	}
 
-	peer->have_sigs[REMOTE] = true;
-	billboard_update(peer);
+	/* If we received remote announcement before, now we need to make
+	 * sure the new announcement is same as the old and then ignore it.
+	 */
+	if(peer->have_sigs[REMOTE] == true) {
+
+		/* make sure new short_channel_id must equal the old one */
+		if (!short_channel_id_eq(remote_scid,
+				 	&peer->short_channel_ids[REMOTE]))
+			peer_failed(&peer->cs,
+				    &peer->channel_id,
+			    	    "Wrong announcement: "
+				    "first received short_channel_ids: %s"
+			    	    " , now second get %s",
+			    	    type_to_string(peer, struct short_channel_id,
+						   &peer->short_channel_ids[REMOTE]),
+			    	    type_to_string(peer, struct short_channel_id,
+						   remote_scid));
+
+		/* make sure new node signature and bitcoin signature equal
```

Comment:
Shouldn't we just check that the signature is _valid_? Signatures change when re-signing (using different k [value](https://en.wikipedia.org/wiki/Elliptic_Curve_Digital_Signature_Algorithm)).

Reply:
Thank you! You're right, and there shouldn't compare with the old sigs if the remote generate new sigs. Gossipd will verify the announcement, so I think we'd better update the new sigs here.